viewmodel-editor/viewmodel-editor.js

```diff
@@ -165,17 +165,24 @@ export default class ViewmodelEditor extends StacheElement {
 
 					listenTo("tagName", () => {
 						Reflect.offValue(serializedJSON, setPatches);
-						json.update({});
+						json.updateDeep({});
 						Reflect.onValue(serializedJSON, setPatches);
-						this.jsonEditorPatches = [];
 					});
 				}
 			},
 
 			jsonEditorPatches: {
-				type: type.ObservableArray,
+				type: type.convert(ObservableArray),
+				value({ lastSet, listenTo, resolve }) {
+					listenTo(lastSet, resolve);
+
+					listenTo("tagName", () => {
+						lastSet.get().updateDeep([]);
```

Comment:
Instead of using `get default() {}` and updating lastSet throughout, it might be easier to follow if you just use a local variable like

```js
let patches = resolve( new ObservableArray([]) )
```